cypress/integration/cucumber/step_definitions/refapp-3.x/02-registration/registration.js

```diff
@@ -0,0 +1,109 @@
+import {Given} from 'cypress-cucumber-preprocessor/steps';
+
+Given('the user login to the Registration Desk', () => {
+    cy.login();
+})
+
+When('the user search for {string}', patientName => {
+    cy.get('button[name=SearchPatientIcon]').click();
+    cy.getByPlaceholder('Search for a patient by name or identifier number').type(patientName);
+})
+
+Then('the result should be {string}', result => {
+    cy.contains(result);
+});
+
+
+When('the user clicks on the add patient icon', () => {
+    cy.get('button[name=AddPatientIcon]').click();
+})
+
+When('the user enters {string} details for Andria Faiza', validity => {
+    const details = {
+        right: {
+            givenName: 'Andria',
+            middleName: 'Kumbha',
+            familyName: 'Faiza',
+            gender: 'Female',
+            birthday: 'January 3, 1970',
+            address: '55372,test',
+            cityVillage: 'Nairobi',
+            stateProvince: 'Nairobi',
+            country: 'Kenya',
+            postalCode: '00100',
+            phoneNumber: '+211 34 567890'
+        },
+        wrong: {
+            givenName: null,
+            middleName: 'Kumbha',
+            familyName: 'Mwangi',
+            gender: 'Female',
+            birthday: 'January 3, 1970',
+            address: null,
+            cityVillage: null,
+            stateProvince: null,
+            country: null,
+            postalCode: null,
+            phoneNumber: null
+        }
+    };
+    // Chek for a valid validity parameter
+    if (!details.hasOwnProperty(validity)) {
+        throw new Error(`Validity '${validity}' is not supported`);
+    }
+    const user = details[validity];
+    if (user.givenName != null) {
+        cy.getByLabel('Given Name').type(user.givenName, {force: true});
```

Comment:
the force option is there because sometimes an alert in the footer covers the input elements. Otherwise, cypress ignores interacting with covered elements.

Reply:
Please add this information as a comment above this code